test/distributed/tensor/test_tensor_ops.py

```diff
@@ -708,6 +709,46 @@ def test_slice(self):
         self.assertEqual(sharded_out.full_tensor(), global_out)
         self.assertEqual(sharded_dtensor.grad.full_tensor(), global_tensor.grad)
 
+    @with_comms
+    def test_split_on_partial(self):
+        self.run_subtests(
+            {
+                "reduce_op": ["sum", "avg", "product", "min", "max"],
+                "split_size": [2, 3, 4],
+                "split_dim": [0, 1],
+            },
+            self._test_split_on_partial,
+        )
+
+    def _test_split_on_partial(self, reduce_op: str, split_size: int, split_dim: int):
+        torch.manual_seed(self.rank)
+        mesh = init_device_mesh(self.device_type, (self.world_size,))
+
+        partial_tensor = torch.randn(8, 8, device=self.device_type)
+        replicate_tensor = partial_tensor.detach().clone()
+        replicate_tensor = funcol.all_reduce(
```

Comment:
A more generic testing thought: it seems to me that this sort of "do the operation with dtensor but also do it with plain tensor" feels like it would be pretty common, can we not automate it?  For example, we don't have to manually setup the replicate_tensor and collective: instead, we can simply do partial_dt.full_tensor().split(split_size, dim=split_dim). We're simply testing that we check this diagram commutes:

```
             split()
  partial_dt ---------> B
      |                  |
      |                  | full_tensor()
full_tensor()       result
      |                  |
      v                  v
      C ---------> result
             split()
```

Reply:
Good point, will add a base PR to introduce such test util. IIRC we have similar logics in each separate test file but haven't merged them into one shared util.

Reply:
implemented @ezyang's suggestion in https://github.com/pytorch/pytorch/pull/158112